src/functional/ffront/decorator.py

```diff
@@ -0,0 +1,294 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+import collections
+import dataclasses
+import functools
+import types
+import typing
+from typing import Any, Optional, Protocol
+
+from eve.utils import UIDs
+from functional.common import GTTypeError
+from functional.ffront import common_types as ct
+from functional.ffront import field_operator_ast as foast
+from functional.ffront import program_ast as past
+from functional.ffront import symbol_makers
+from functional.ffront.foast_to_itir import FieldOperatorLowering
+from functional.ffront.func_to_foast import FieldOperatorParser
+from functional.ffront.func_to_past import ProgramParser
+from functional.ffront.past_passes.type_deduction import ProgramTypeDeduction
+from functional.ffront.past_to_itir import ProgramLowering
+from functional.ffront.source_utils import ClosureRefs
```

Comment:
This is the convention we used elsewhere and having another convention just because the amount of imports exceeds a limit doesn't make sense to me (I don't like this style of importing and naming anyway). The problem is another one here: This file contains too many different components and should be split up. I've added a comment.

Reply:
I'm not aware of any convention about directly importing objects instead of modules (except for typing classes) but you're right this is not  documented anywhere yet.